### ISSUE TITLE
can't apply patch to init.rc

### ISSUE THREAD

**Issue by trusiad** [micgeri/zte-openc-switch-bluedroid#1]
I tried the script with boot.img. 
the script stops with this message : http://pastebin.com/XB5S7bAX
there's no init.rc file in the folder, how come ?
regards, 
trusiad

**Comment by micgeri** [comment]
Hello trusiad,

Normally, the script get init.rc in the boot.img file. Please, can you list the file in /home/david/zte-openc-switch-bluedroid-master/update-bootimage/tmp/boot/ramdisk ?

**Comment by trusiad** [comment]
The boot.img comes from zte. J'ai suivi le lien de téléchargement de la page http://builds.firefoxos.mozfr.org/doc/fr/devices/zte-open-c-fr.L'adresse est la suivante : http://www.ztefrance.com/downloads/Pack_root_du_ZTE_Open_C.zip. 
The ramdisk folder is empty. Nothing in it...
regards, 
trusiad

**Comment by micgeri** [comment]
Ok. I used a root pack I downloaded a long time ago. I'm not at home, and I have a limited mobile connection, so I can't download "Pack root" now. I will try to download the pack this evening, and test the script with its boot.img file.

**Comment by micgeri** [comment]
Hello trusiad,

I can't reproduce your issue with the boot.img. However, I updated several parts in the scripts about the base directory, and the mkbootimg_tools version.
Can you get the last version of the script and retry ?

Please keep me track of the result.

Best regards,
micgeri

**Comment by trusiad** [comment]
That's weard. Are you using the root pack from zte ?
I just tried with the new version, nut the init.rc file is still missing.
***
./update-bootimage.sh 
--2015-11-30 07:38:42--  https://github.com/micgeri/mkbootimg_tools/archive/bootimg-openc-ok.zip
Résolution de github.com (github.com)… 192.30.252.129
Connexion à github.com (github.com)|192.30.252.129|:443… connecté.
requête HTTP transmise, en attente de la réponse… 302 Found
Emplacement : https://codeload.github.com/micgeri/mkbootimg_tools/zip/bootimg-openc-ok [suivant]
--2015-11-30 07:38:44--  https://codeload.github.com/micgeri/mkbootimg_tools/zip/bootimg-openc-ok
Résolution de codeload.github.com (codeload.github.com)… 192.30.252.144
Connexion à codeload.github.com (codeload.github.com)|192.30.252.144|:443… connecté.
requête HTTP transmise, en attente de la réponse… 200 OK
Taille : non indiqué [application/zip]
Sauvegarde en : « /home/david/zte-openc-switch-bluedroid-master/update-bootimage/tmp/mkboot.zip »

/home/david/zte-ope     [      <=>             ]   3,92M   439KB/s   ds 10s    

2015-11-30 07:38:56 (389 KB/s) - « /home/david/zte-openc-switch-bluedroid-master/update-bootimage/tmp/mkboot.zip » sauvegardé [4109445]

Archive:  /home/david/zte-openc-switch-bluedroid-master/update-bootimage/tmp/mkboot.zip
07b305461d03a5b9f60cea0a969fec16f6e8270a
   creating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/
   creating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/README.md  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/bash  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/cpio  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/file  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/grep  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/gzip  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/lz4  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/lzma  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/lzop  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/magic.mgc  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/mkboot  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/mkbootfs  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/mkbootimg  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/od  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/wrapper  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/ARM/xz  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/README.md  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/dtbTool  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/dtbToolCM  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/dtbtool.txt  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/dtc  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/lz4  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/mkboot  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/mkbootfs  
  inflating: /home/david/zte-openc-switch-bluedroid-master/update-bootimage/mkbootimg_tools-bootimg-openc-ok/mkbootimg  
Unpack & decompress /home/david/zte-openc-switch-bluedroid-master/update-bootimage/put-files-here/boot.img to /home/david/zte-openc-switch-bluedroid-master/update-bootimage/tmp/boot
  kernel         : kernel
  ramdisk        : ramdisk
  page size      : 2048
  kernel size    : 5064720
  ramdisk size   : 1154233
  dtb size       : 1169408
  base           : 0x00000000
  kernel offset  : 0x00008000
  ramdisk offset : 0x02000000
  tags offset    : 0x01e00000
  dtb img        : dt.img
  cmd line       : console=ttyHSL0,115200,n8 androidboot.console=ttyHSL0 androidboot.hardware=qcom user_debug=31 msm_rtb.filter=0x37
ramdisk is gzip format.
Unpack completed.
can't find file to patch at input line 3
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|--- a/boot/ramdisk/init.rc	1969-12-31 20:00:00.000000000 -0400
|+++ b/boot/ramdisk/init.rc	2015-11-22 14:37:31.971823980 -0400
--------------------------
File to patch:

**Comment by micgeri** [comment]
Yes, I forgot to precise it, but I downloaded yesterday the root pack from ZTE France website, got its boot.img file, and it worked fine.
Can you temporarily upload your boot.img file for me please ?

Also, can you verify the md5 checksum of your boot.img file ? You can do this with : `md5sum  /home/david/zte-openc-switch-bluedroid-master/update-bootimage/put-files-here`

Finally, can you test these commands to extract ramdisk.packed file :
```
cd /home/david/zte-openc-switch-bluedroid-master/update-bootimage/tmp/boot
mkdir ramdisk_extr
cd ramdisk_extr
gunzip -c ../ramdisk.packed | cpio -i
```
After executing these commands, is the ramdisk file extracted in __ramdisk_extr__ folder, and is it a file named __init.rc__ ?

Best Regards,
micgeri

**Comment by trusiad** [comment]
cpio ! that's it ! works fine now.

**Comment by micgeri** [comment]
Ok, but can you still send me an md5 checksum of your boot.img file please, just to be sure.

**Comment by trusiad** [comment]
1eefe655aa9cabd7103b7e0c424b4806
Voilà. Merci pour le coup de pouce.

**Comment by micgeri** [comment]
Ok merci, je voulais juste vérifier que le fichier n'était pas corrompu ou altéré.
De rien. J'attends d'autres retours, et si le problème se reproduit, je modifierai certainement le script pour inclure cette commande.

**Comment by trusiad** [comment]
Encore moi. 
J'ai mon fichier, mais le second script pose aussi problème. 

./switch-openc-to-bluedroid.sh : Extraction de l'archive contenant les premiers fichiers pour bluedroid...
./switch-openc-to-bluedroid.sh : Téléchargement et extraction si nécessaire de l'archive Kitkat pour Open C...
./switch-openc-to-bluedroid.sh : Montage de l'image system.img sur un dossier temporaire (le mot de passe root peut être requis pour la commande 'umount')...
./switch-openc-to-bluedroid.sh : Récupération des blobs...
./switch-openc-to-bluedroid.sh : Démontage de l'image system.img (le mot de passe root peut être requis pour la commande 'umount')...
./switch-openc-to-bluedroid.sh : Envoi des fichiers sur le téléphone...
failed to copy '/home/david/zte-openc-switch-bluedroid-master/switch-openc-to-bluedroid/put-files-here/boot.img' to '/storage/sdcard/boot_bth.img': Read-only file system
rm failed for /storage/sdcard/boot_bth.img, No such file or directory

**Comment by micgeri** [comment]
Je suppose que ton téléphone est rooté.
Peux-tu me dire ce que contient le dossier storage sur ton téléphone ? `adb shell ls -l /storage`

**Comment by trusiad** [comment]
dr-xr-xr-x root     root              1970-08-22 12:01 emulated
drwxrwxrwx system   sdcard_rw          1970-01-01 01:00 sdcard
lrwxrwxrwx root     root              1970-08-22 12:01 sdcard0 -> /storage/emulated/legacy
drwxrwxrwx system   sdcard_rw          1970-01-01 01:00 sdcard1

**Comment by micgeri** [comment]
Maintenant, que donne `adb shell ls -l /storage/sdcard` ?

**Comment by trusiad** [comment]
drwxrwxrwx system   sdcard_rw          2015-11-29 15:57 DCIM
drwxrwxrwx system   sdcard_rw          1980-01-01 00:00 LOST.DIR

**Comment by micgeri** [comment]
Ok. Donnes-moi le résultat des deux commandes suivantes :
- `adb shell touch /storage/sdcard/testfile` (à priori le résultat doit être le même -readonly-, mais bon)
- `adb shell touch /storage/sdcard1/testfile`

**Comment by trusiad** [comment]
Rien du tout. Rien de visible/lisible. Aucun message en retour.

**Comment by micgeri** [comment]
Désolé, j'ai mis à jour mon poste précédent.

**Comment by trusiad** [comment]
[david@bureau ~]$ adb shell touch /storage/sdcard/testfile
[david@bureau ~]$ adb shell touch /storage/sdcard1/testfile
[david@bureau ~]$

**Comment by micgeri** [comment]
Tu as oublié une partie des commandes. Ce n'est pas grave. Dis-moi simplement que retournent `adb shell ls -l /storage/sdcard` et `adb shell ls -l /storage/sdcard1` ?

**Comment by micgeri** [comment]
Si tu as un fichier testfile sur les 2, il faudra que j'investigue un peu plus, car je commence à être à court d'idée.

**Comment by trusiad** [comment]
Cela ne donne rien. Le but de la manoeuvre était de créer un fichier, et de vérifier qu'il était bien là ensuite ? Il n'y a rien...
[david@bureau ~]$ adb shell touch /storage/sdcard/testfile
[david@bureau ~]$ adb shell ls -l /storage/sdcard
[david@bureau ~]$ 
Si je comprends bien, le fichier testfile se perd quelque part...

**Comment by micgeri** [comment]
Exact, c'était bien le but. Et oui, le fichier est bien perdu quelque part.

Il y a un truc de bizarre, car dans ton post plus haut, tu m'as dit que la commande `adb shell ls -l /storage/sdcard` t'a renvoyé DCIM et LOST.DIR, tandis que là, tu n'as rien.
Tu es sûr de ne pas avoir mis sdcard au lieu de sdcard1 (ou l'inverse) ?

**Comment by trusiad** [comment]
Non. Je viens de me rendre compte exactement de la même chose.

**Comment by micgeri** [comment]
Précision : sdcard correspond à la mémoire interne de l'Open C, tandis que sdcard1 correspond à une microsd

**Comment by micgeri** [comment]
Que te donne la commande `df` ?

**Comment by trusiad** [comment]
Oui, j'ai repéré, grâce aux dossiers. Je regarde, j'essaie de comprendre, et je reviens vers toi. 
Bonne fin de soirée, 
trusiad

**Comment by micgeri** [comment]
Ok, oui tiens-moi au courant stp, car les problèmes que tu rencontres m'intriguent.
Merci, de même.

micgeri.

**Comment by trusiad** [comment]
adb usb && adb remount.
Ça a marché. J'ai pu faire tourner le second script. Mais je n'ai toujours pas de bluetooth... J'ai une version nightly. Il y a autre chose à faire ?
Bonne soirée, 
pour de bon.
Trusiad

**Comment by micgeri** [comment]
Ok pour adb remount. Je ne le faisais qu'avant l'envoi des fichiers sur la partition system, car avec mon téléphone, j'arrive à écrire sur les 'sdcard' sans le faire. J'ai adapté le script en conséquence.

Si tu es sur la branche nightly des builds communautaires, c'est normal. Ces builds jusqu'ici ne prennent pas en charge bluedroid. Il faudrait que tu suives [ce sujet](https://github.com/mozfr/buildscommunautaires/issues/2), afin de participer au test de la build avec prise en charge de bluedroid.
**N'oublies pas d'effectuer une sauvegarde avant d'installer la build.**

micgeri

**Comment by micgeri** [comment]
Bonjour trusiad,

Du nouveau concernant ton problème ?
As-tu pu le résoudre ?

micgeri

**Comment by trusiad** [comment]
Oui. J'ai installé la Spark, et le bluetooth fonctionne !
Le problème est donc résolu. 
Merci beaucoup, 
trusiad

**Comment by micgeri** [comment]
Ok de rien, et merci du retour.
Peux-tu aussi remplir le formulaire de test pour cette build stp ? Le lien est présent dans le sujet que je t'ai envoyé précédemment.

micgeri

**Comment by micgeri** [comment]
resolved #1
